### PR TITLE
Release/run ya provider script

### PR DESCRIPTION
**Pull request by tworec**
script for installing locally, configuring and running yagna with ya-provider with two runtimes:
- WASI
- VM

### CODE REVIEW

File: agent/provider/run-ya-provider.sh [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    if [[ ! -d "$install_dir" ]]; then
+        say "Creating install dir $install_dir"
+        ensure mkdir "$install_dir"
+    else
+        say "Using install dir $install_dir"
+    fi
+
+    cd "$install_dir"
+
+    if [[ ! -f usr/lib/yagna/plugins/ya-runtime-wasi.json ]]; then
```
Comment (mfranciszkiewicz) [inline agent/provider/run-ya-provider.sh:91]:
```suggestion
    if [[ ! -f /usr/lib/yagna/plugins/ya-runtime-wasi.json ]]; then
```

File: agent/provider/run-ya-provider.sh [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    	  say "Install WASI runtime"
+        ensure wget https://github.com/golemfactory/ya-runtime-wasi/releases/download/v0.2.0/ya-runtime-wasi_0.2.0_amd64.deb
+        dpkg-deb -R ya-runtime-wasi_0.2.0_amd64.deb .
+        rm -rf DEBIAN
+    fi
+
+
+    if [[ ! -f usr/lib/yagna/plugins/ya-runtime-vm.json ]]; then
```
Comment (mfranciszkiewicz) [inline agent/provider/run-ya-provider.sh:99]:
```suggestion
    if [[ ! -f /usr/lib/yagna/plugins/ya-runtime-vm.json ]]; then
```

File: agent/provider/run-ya-provider.sh (hunk 1 above)
Comment (tworec) [reply agent/provider/run-ya-provider.sh:91]: nope this is intentional

**Review by mfranciszkiewicz — DISMISSED** [review]

File: agent/provider/run-ya-provider.sh [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    if [[ ! -d "$install_dir" ]]; then
+        say "Creating install dir $install_dir"
+        ensure mkdir "$install_dir"
+    else
+        say "Using install dir $install_dir"
+    fi
+
+    cd "$install_dir"
+
+    if [[ ! -f usr/lib/yagna/plugins/ya-runtime-wasi.json ]]; then
```
Comment (zakaprov) [inline agent/provider/run-ya-provider.sh:136]: Perhaps leave a comment here about this path being intentional, as it seems to be confusing (I had the same question about this part).

File: agent/provider/run-ya-provider.sh [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+    # TODO: kill service when this script exits
+    say "Starting Yagna Service... (stdout & err in $(pwd)/yagna.log)"
+    RUST_LOG=debug ../usr/bin/yagna --accept-terms service run >> yagna.log 2>&1 &
+    local pid="$!"
+    echo "$pid" > "$pid_file"
+    sleep 2s # wait a bit for service to start fully
+    say "Yagna Service started ($pid)."
+
+    if grep "__GENERATED_APP_KEY__" .env > /dev/null; then
```
Comment (zakaprov) [inline agent/provider/run-ya-provider.sh:192]:
```suggestion
    if grep -q "__GENERATED_APP_KEY__" .env; then
```

**Review by zakaprov — DISMISSED** [review]
Left two minor comments, looks good otherwise.

**Comment by tworec** [comment]
code moved to separate repository [golemfactory/yagna-testnet-scripts](https://github.com/golemfactory/yagna-testnet-scripts)

**Pull request closed without merging**